src/util/Recoil_deepFreezeValue.js

```diff
@@ -56,6 +56,10 @@ function shouldNotBeFrozen(value: mixed): boolean {
     return true;
   }
 
+  if (!!value && value === value.window) {
```

Comment:
Why is it checking for equality of a property called `window`?  Seems like it should check for equality with the global `window`.  Though, for SSR compatibility, it would need some `typeof window !== 'undefined'` check which has been added to the codebase in a few locations.  We should probably normalize that out at this point, especially if it would be used in a loop here.

Reply:
@drarmstr 
>Why is it checking for equality of a property called window?
sorry, this is hard to understand.
I fix it to use `typeof window !== undefined && value instanceof Window` instead of above.